modules/core/src/main/scala/fs2/kafka/AdminClientSettings.scala

```diff
@@ -25,7 +23,7 @@ import scala.concurrent.duration._
   * Use [[AdminClientSettings#apply]] for the default settings, and
   * then apply any desired modifications on top of that instance.
   */
-sealed abstract class AdminClientSettings[F[_]] {
+sealed abstract class AdminClientSettings {
```

Comment:
With this change, I wonder why we can't use common case class here? The current solution provides almost the same functionality but with tons of boilerplate inside the library.

Reply:
I agree it's boilerplatey,but the problem with using case classes directly for config is that adding fields breaks binary compatibility.

One thing we could do to reduce boilerplate, though, is implement the `withProperty`-based methods directly in the abstract class rather than in the implementation class.

Reply:
Oh, I forgot about this problem with case classes.